src/vellum/client/core/client_wrapper.py

```diff
@@ -20,7 +28,8 @@ def get_headers(self) -> typing.Dict[str, str]:
             "X-Fern-SDK-Name": "vellum-ai",
             "X-Fern-SDK-Version": "0.14.53",
         }
-        headers["X-API-KEY"] = self.api_key
+        headers["X-API-KEY"] = self._workspace_api_key
+        headers["X-API-KEY"] = self.environment_api_key
```

Comment:
This is the biggest problem to solve. Right now, every API is using the environment api key. We will need something like:

```python
def get_headers(self, scheme) -> typing.Dict[str, str]:
    ...
    if scheme == "environmentApiKeyAuth":
        headers["X-API-KEY"] = self.environment_api_key
    elif scheme == "workspaceApiKeyAuth":
        headers["X-API-KEY"] = self.workspace_api_key
```

then, thread it through from the client. Am realizing mid-comment that it will be easier for me to communicate this via PR, which I'll branch off this to do now

Reply:
https://github.com/vellum-ai/vellum-python-sdks/pull/1625/files